QEfficient/generation/text_generation_inference.py

```diff
@@ -235,111 +266,112 @@ def cloud_ai_100_exec_kv_helper(
     end = perf_counter()
     generated_texts = tokenizer.batch_decode(generated_ids, skip_special_tokens=True)
 
-    for i in range(1 if stream else 0, batch_size):
-        print()
-        print(i, prompt[i], generated_texts[i])
+    if stream:
+        for i in range(1, batch_size):
+            print("\n\n=====================================================================\n")
+            print("Prompt : ", prompt[i])
+            print("Completion :", generated_texts[i])
+        print("\n\n=====================================================================\n")
 
     prefill_time = loop_start - start
     decode_perf = (num_token - 1) / (end - loop_start)
     total_perf = num_token / (end - start)
     total_time = end - start
     print()
 
-    latency_stats = (generated_texts, prefill_time, decode_perf, total_perf, total_time)
-    return latency_stats
+    return CloudAI100ExecInfo(
+        generated_texts=generated_texts,
+        generated_ids=generated_ids,
+        prefill_time=prefill_time,
+        decode_perf=decode_perf,
+        total_perf=total_perf,
+        total_time=total_time,
+    )
 
 
-def print_latency_stats_kv(
-    prompt, generated_texts, batch_size, prefill_time, decode_perf, total_perf, total_time, automation: bool = False
-):
+def print_latency_stats_kv(prompt, batch_size, execinfo, automation: bool = False):
     if automation:
         print()
         print("input=", prompt)
-        print("output=", generated_texts)
-        print("Prefill time a.k.a TTFT is=", round(prefill_time, 2))
-        print("Decode token/sec is=", round(decode_perf * batch_size, 2))
-        print("Total token/sec is=", round(total_perf * batch_size, 2))
-        print("Total (E2E) inference time is=", round(total_time, 2))
+        print("output=", execinfo.generated_texts)
```

Comment:
Just do `print(execinfo)` here and move this code in `__repr__` method of the dataclass